airbyte-cdk/python/airbyte_cdk/sources/message/repository.py

```diff
@@ -62,10 +72,47 @@ def emit_message(self, message: AirbyteMessage) -> None:
             raise ValueError(f"As of today, only {_SUPPORTED_MESSAGE_TYPES} are supported as part of the InMemoryMessageRepository")
         self._message_queue.append(message)
 
-    def log_message(self, level: Level, message_provider: Callable[[], AirbyteMessage]) -> None:
-        if self._SEVERITY_BY_LOG_LEVEL[self._log_level] >= self._SEVERITY_BY_LOG_LEVEL[level]:
-            self.emit_message(AirbyteMessage(type=Type.LOG, log=AirbyteLogMessage(level=level, message=message_provider())))
+    def log_message(self, level: Level, message_provider: Callable[[], LogMessage]) -> None:
+        if _is_severe_enough(self._log_level, level):
+            self.emit_message(
+                AirbyteMessage(type=Type.LOG, log=AirbyteLogMessage(level=level, message=filter_secrets(json.dumps(message_provider()))))
+            )
 
     def consume_queue(self) -> Iterable[AirbyteMessage]:
         while self._message_queue:
             yield self._message_queue.pop(0)
+
```

Comment:
I doubt we're going to store enough messages to make this a big deal, but fyi `_message_queue` can be a deque instead of a list, to give it an efficient left-pop operation (sorry, missed when I reviewed the original message repository code).